bitcoin/src/consensus/encode.rs

```diff
@@ -207,33 +206,33 @@ pub trait ReadExt : io::Read {
 }
 
 macro_rules! encoder_fn {
-    ($name:ident, $val_type:ty, $writefn:ident) => {
+    ($name:ident, $val_type:ty) => {
         #[inline]
         fn $name(&mut self, v: $val_type) -> Result<(), io::Error> {
-            self.write_all(&endian::$writefn(v))
+            self.write_all(&v.to_le_bytes())
         }
     }
 }
 
 macro_rules! decoder_fn {
-    ($name:ident, $val_type:ty, $readfn:ident, $byte_len: expr) => {
+    ($name:ident, $val_type:ty, $byte_len: expr) => {
         #[inline]
         fn $name(&mut self) -> Result<$val_type, Error> {
             bitcoin_internals::const_assert!(core::mem::size_of::<$val_type>() == $byte_len);
             let mut val = [0; $byte_len];
```

Comment:
Actually, `[0; core::mem::size_of::<$val_type>()]` should be possible in our MSRV because `size_of` is `const fn`. So you could remove the parameter and assert completely. The assert is also pointless anyway since the type is now checked by the signature of `from_le_bytes`.

Reply:
So it is, cheers. I remove the assertion all together.